src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -881,7 +879,25 @@ internal async Task Close(bool connectionClosing, bool async, bool isDisposing)
             case ConnectorState.Executing:
             case ConnectorState.Connecting:
                 if (State != ReaderState.Consumed)
-                    await Consume(async);
+                {
+                    try
+                    {
+                        await Consume(async);
+                    }
+                    catch (TimeoutException)
+                    {
+                        // nothing to do here
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        // nothing to do here
+                    }
+                    catch
+                    {
+                        Debug.Assert(Connector.IsBroken);
+                        throw;
+                    }
```

Comment:
Let's simplify it:
```suggestion
                    catch (Exception ex) when (
                        ex is TimeoutException ||
                        ex is OperationCancelledException)
                    {
                        // nothing to do here
                    }
                    catch
                    {
                        Debug.Assert(Connector.IsBroken);
                        throw;
                    }
```

Or rewrite the condition with some sugar (but I prefer the variant above):
````csharp
                    catch (Exception ex) when (ex is TimeoutException or OperationCancelledException)
```

Checking the type in the last catch is fine too, so decide what you like more.